v2/internal/genericarmclient/policy_register_rp.go

```diff
@@ -0,0 +1,221 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package genericarmclient
+
+// NOTE: This file was copied from https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/arm/runtime/policy_register_rp.go
+// with modifications. The primary modification was to remove all retries and immediately report an error after registration.
+// That error will be categorized and retried on by the operator.
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	armpolicy "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm/policy"
+	armruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+	azpolicy "github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+
+	"github.com/Azure/azure-service-operator/v2/internal/version"
+)
+
+// NewRPRegistrationPolicy creates a policy object configured using the specified options.
+// The policy controls whether an unregistered resource provider should automatically be
```

Comment:
Do we want to automatically register the RP? Thinking if it would be a right thing to do without the user's permission?

Reply:
So it's pretty common to automatically register subscriptions on the users behalf. The Go SDK (and AFAIK all of the other SDKs) actually do it by default. We turn it off on our usage of the Go SDK because it also by default waits 5m for the registration to take place, which blocks the running reconcilers for 5m which is bad.

There's not really a lot of risk/harm in registering an RP AFAIK. We also have a user requesting it in #2513, and not registering sorta defeats some of the IAC purpose of ASO because you have to perform manual actions to initially set up your sub. In a perfect world ASO would take care of that for you.